src/hb-open-type.hh

```diff
@@ -209,7 +209,9 @@ struct Offset : Type
   void *serialize (hb_serialize_context_t *c, const void *base)
   {
     void *t = c->start_embed<void> ();
-    c->check_assign (*this, (unsigned) ((char *) t - (char *) base));
+    c->check_assign (*this,
+                     (unsigned) ((char *) t - (char *) base),
+                     HB_SERIALIZE_ERR_OTHER);
```

Comment:
This should be `ERR_OFFSET_OVERFLOW`, right? This is the one case we talked about.

Reply:
Yes this is the case, so offset overflow is not treated as a fatal error and since this overflow happens outside of the add_link/resolve_links mechanism the repacker can't deal with it so we need to treat it as a fatal error by instead using ERR_OTHER. I could maybe instead add a ERROR_OFFSET_OVERFLOW_FATAL?

Reply:
I suggest considering all errors the same. Just that **inside resolve_links** if you start with no error and end up with offset-overflow then you know you have repacking to do.